### DISCUSSION TITLE
Dark Dungeon

### DISCUSSION THREAD

**Discussion by Internetzspacezshipz** [CyberSkull/Godot-Wild-Jam-72#2] [Ideas]
Primary theme:
![MainTheme](https://github.com/user-attachments/assets/306a1b62-0cb1-4192-b59e-71ca34ca0452)

Optional wildcards:
![Wildcards](https://github.com/user-attachments/assets/7f2cca0d-ffa7-4e63-b752-7774207d089e)

**Top Level Description:**
A top-down dungeon-crawler-roguelike type game, where your character has to move and activate light sources.

Enemies gain bonuses when inside the darkness, so it is in your best interest to stay in the light and change the lighting environment to gain an advantage. Your character cannot carry light sources while also fighting enemies. Some enemies have ranged attacks from the darkness, so you might need to use level elements to light them up so that they do not have buffs.

**Wildcard Usages:**
(Wildcard #2) There are items scattered around the level that randomly can give either you or the enemies buffs.
Buffs for enemies can include new attack types, special abilities to temporarily lower the strength/distance of your light sources, or passive bonuses like increased damage, health, movement speed.

(Wildcard #3) I am sure we can put a duck in somewhere, possibly as an enemy type, or as a general theme to the world (maybe the floors have ducks etched into them!). Perhaps this is the “Dungeon of the Immortal Duck”... (could be a boss enemy perhaps? The final boss?)

**Powerups**
Buffs for the player can include things like throwable temporary light sources, special attacks for countering specific situations/enemy types (shields, or magic force-fields for defending areas/objects in the level from enemy attacks), increasing light source strength/distance, or passive bonuses like increased damage, health, or movement speed. There could also be the possibility of powerups like causing enemies to take damage while in the light.

**Light system:**
Since the light would be the main mechanic of this system, there should be multiple ways and objects to interact with it:
In level static objects:
Mirrors - allows light to bounce around corners
Static light sources
Spotlight - light points out in a cone from the object
Omnidirectional lights - light goes in all directions around the object (shorter distance than the spotlight)
Lenses - able to focus light from light sources to have further reach
In order to best utilise these mechanics, the level design must follow suit.

**Enemies:**
3-4 different types of enemies with a couple different abilities.
Melee enemy: attacks only at close range, usually tries to stay in the dark unless the player is much closer than the darkness is
Ranged enemy: shoots arrows from the darkness. Will always try to be in darkness at all costs.
Debuff enemy: throws smoke bombs (could be other things too, idk) to block light from entering certain regions so they can move to attack you. When in light, will attempt to run into darkness or throw their light blocking item to stop the light from affecting them or allies. Has a close range melee attack (lower damage output than normal melee enemies)
Long melee enemy: attacks with a long spear or other long melee weapon. Being inside the light makes them unable to use their long spear, and they can only fight at close range.

**Enemy Behaviour:**
The enemies attempt to stay in the darkness for the most part if they can. If they are caught out in the light with the closest darkness being some multiple of the distance to the player, then they will instead just attempt to attack. This could be some different values for different types of enemies. Eg. the melee enemy attacks the player regardless of light level only if the nearest darkness is 2 or more times the distance to the player. Eg2. the ranged enemy attacks regardless of light level only when the nearest darkness is 4 or more times the distance to the player.

**Level design:**
Levels could be randomly generated or designed from scratch.

The design of the levels should be rooms with walls, pillars and such to block lines of sight/light, so that enemies can hide around close corners in the darkness, waiting for you to come nearby to attack you. Since you cannot move your light sources while also fighting, it means you need to strategically place your light sources to attempt to debuff/reveal enemies.

**Player:**
The player is a guy with a sword/other melee weapon raiding this dungeon.
Not 100% sure how the character’s attacks should work or how in depth the character’s combat abilities should be though. Definitely would be subject to testing gameplay.
The player should probably have superior movement speed compared to the enemies in order to allow the player to take advantage of changing the light, and causing the enemies to move to darkness.

**Strategies employable by the player to win fights more easily than fighting in darkness:**
Divide and conquer: the player can place lights in such a way to cause enemies to split up so they cannot attack you at the same time, forcing 1v1 fights between the player and enemies
Forcing enemies into a pocket and avoiding them to completely bypass them. Enemies do not always need to be defeated to succeed in a level.
Forcing enemies into light so they fight in a massively debuffed state, even taking 1v3 fights should be easy enough for the player due to the enemy debuffs while in the light!

**Simple level gameplay example:**

![GameIdeaImage0](https://github.com/user-attachments/assets/91810f38-54d9-4576-8045-e38d8d645380)
![GameIdeaImage1](https://github.com/user-attachments/assets/7d6e6481-46b3-4c0b-b0cd-f094fb4d0213)

Finish option A:
![GameIdeaImage2](https://github.com/user-attachments/assets/cb736a0b-9745-4c34-b7d3-cf60bfa0dce8)
Finish option B:
![GameIdeaImage3](https://github.com/user-attachments/assets/16a0c3b9-9e12-419c-acfd-4f9c6212fde7)

**Comment by CyberSkull** [comment]
https://docs.google.com/document/d/10iIuY10_H25JpPuGTPwvYvNa8YPdun2npG8JguPPquc/edit
